lib/jobs/obm-control.js

```diff
@@ -64,18 +64,33 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
+            //var settings = _.find(node.obmSettings, function(setting) {
+            self.settings = _.find(node.obmSettings, function(setting) {
                 return setting ? setting.service === self.obmServiceName : false;
             });
 
-            assert.ok(settings, "Node should have OBM settings for service: " +
+            if(!self.settings){
```

Comment:
(1) In some situation, we are aware some obm services will not work, for example, snmp-obm-service works for ipdu/switch, ipmi-obm-service does't work, so we want to explicitly use that particular obm-service. 
(2) For passive node discovery, such as the discovery for management service via ipmi, the discovery for switch via snmp, only that particular obm-service is reliable. 
(3) As I know, only ipmi-obm-service supports set pxe boot, so if I want to set the pxe boot order, I should explicitly use the ipmi-obm-service, otherwise fails.